File: donuts/donut.js
            let A=0,B=0,M=
        Math;const a=()=>{let s
      =[],t=[];A+=.05,B+=.07;const
    o=M.cos(A),e=M.sin(A),n=M.cos(B
   ),c=M.sin(B);for (let o=0;o<1760;
 o++)s[o]=o%80==79?"\n":" ",t[o]=0;for
(let i=0;i<6.28;i+=.07){const r=M.cos(
i),a=M.sin(i);for(let i=0;i<6.28;i+=.02
){const l=M.sin        (i),f=M.cos(i),A
=r+2,B=1/(l*A*          e+a*o+5),d=l*A*
o-a*e,m=40+30*          B*(f*A*n-d*c)|0
,v=12+15*B*(f*A        *c+d*n)|0,I=m+80
*v,h=8*((a*e-l*r*o)*n-l*r*e-a*o-f*r*c)|
 0;v<22&&v>=0&&m>=0&&m<79&&B>t[I]&&(t[
 I]=B,s[I]=".,-~:;=!*#$@"[h>0?h:0])}}
  process.stdout.write(`\x1b[J\x1b[H`
  +s.join(""))},i=setInterval(a,50
      );/*=!!!**********!!!==:*/
        /*~~;EvanZhouDev;;:~*/
            /*.,-2023--,.*/

File: donuts/donut.dart
             import"dart:math"as
          M; import "dart:io"; void 
       main(){var A=.0,B=.0;while(true
      ){List<String>s=[];List<double>t=
    [];A+=.05;B+=.07;var o=M.cos(A),e= M.
   sin(A),n=M.cos(B),c=M.sin(B);for(var i=
  0;i<1760;i++){s.add(i%80==79?"\n":" ");t.
 add(.0);}for(var i       =.0;i<6.28;i+=.07)
 {var r=M.cos(i),           w=M.sin(i);for (
 var j=.0;j<6.28             ;j+=.02){var l=
 M.sin(j),f = M.             cos(j),g=r+2,q=
 1/(l*g*e+w*o+5)             ,d=l*g*o-w*e,m=
 (40+(30*q*(f*g*n           -d*c))).toInt(),
 v=(12+(15*q*(f*g*c       +d*n))).toInt(),x=
  m+80*v,h=(8*((w*e-l*r*o)*n-l*r*e-w*o-f*r*
   c)).toInt();if(v<22&&v>=0&&m>=0&&m<80&&
    q>t[x]){t[x]=q;s[x]=".,-~:;=!*#\$@"[h
      >0?h:0];}}}stdout.write("\x1b[H"+s
       .join(""));sleep(const Duration
          (milliseconds:50));}}/*..
             Dart..by drakeerv*/